common/remote/sockfile.cpp

```diff
@@ -5626,18 +5870,53 @@ class CRemoteFileServer : implements IRemoteFileServer, public CInterface
             }
             else
             {
-                for (unsigned __int64 i=0; i<defaultDaFSNumRecs; i++)
+                if (grouped)
                 {
-                    size32_t rowSz;
-                    const void *row = outputActivity->nextRow(rowSz);
-                    if (!row)
+                    bool pastFirstRow = outputActivity->queryProcessed()>0;
+                    for (unsigned __int64 i=0; i<defaultDaFSNumRecs; i++)
                     {
-                        eoi = true;
-                        break;
+                        size32_t rowSz;
+                        const void *row = outputActivity->nextRow(outBuilder, rowSz);
+                        if (!row)
+                        {
+                            if (!pastFirstRow)
+                            {
+                                eoi = true;
+                                break;
+                            }
+                            else
+                            {
+                                responseWriter->outputBeginNested("Eog", false);
+                                responseWriter->outputEndNested("Eog");
+                                row = outputActivity->nextRow(outBuilder, rowSz);
+                                if (!row)
+                                {
+                                    eoi = true;
+                                    break;
+                                }
+                            }
+                        }
+                        responseWriter->outputBeginNested("Row", true);
+                        out->toXML((const byte *)row, *responseWriter);
+                        responseWriter->outputEndNested("Row");
+                        pastFirstRow = true;
```

Comment:
Worth resetting outBuilder buffer, so it doesn't keep growing (and in the else below)?

Reply:
oh yes, well spotted.